main.tf

```diff
@@ -10,7 +10,7 @@ module "ssh-key" {
 resource "azurerm_kubernetes_cluster" "main" {
   name                    = var.cluster_name == null ? "${var.prefix}-aks" : var.cluster_name
   kubernetes_version      = var.kubernetes_version
-  location                = data.azurerm_resource_group.main.location
+  location                = var.location ==null ? data.azurerm_resource_group.main.location : var.location
```

Comment:
```suggestion
  location                = var.location == null ? data.azurerm_resource_group.main.location : var.location
```

Reply:
This update was working for me, but curious to see if you are noticing any create/delete cycles still, @larryclaman or @kdouglas-qarik!

Reply:
I ended up dropping use of this module entirely and putting the required terraform resources directly in my own module instead, but based on what I saw previously, setting the location with an optional var should prevent the create/delete cycle yes.